Add Docker project is ready for the code review

misterSokor opened the pull request:
Hi! I just wanted to mention that since my computer is an older iMac (late 2013) running Mojave, it doesn’t support newer versions of Docker. Because of that, the way I set up this project might look a little different from what you expect, especially in the application.properties file. Thank you!

misterSokor committed:
Add Docker project is ready for the code review

Elena-Bruyako reviewed (changes requested):
remove comments
provide .env.template file
add dependency to pom file
Have you tested your app in docker? There is image only for database

misterSokor committed:
- remove unnecessary comments
- add .env.template file for environment variables
- add missing dependency to pom.xml
- note: tested with Docker (only database image is provided)

Elena-Bruyako commented on Dockerfile:
```diff
@@ -0,0 +1,5 @@
+FROM openjdk:17-jdk-slim
```
revise Docker module
this file should look like 
```
# Builder stage
FROM openjdk:17-jdk-alpine as builder
WORKDIR application
ARG JAR_FILE=target/*.jar
COPY ${JAR_FILE} application.jar
RUN java -Djarmode=layertools -jar application.jar extract

# Final stage
FROM openjdk:17-jdk-alpine
WORKDIR application
COPY --from=builder application/dependencies/ ./
COPY --from=builder application/spring-boot-loader/ ./
COPY --from=builder application/snapshot-dependencies/ ./
COPY --from=builder application/application/ ./
ENTRYPOINT ["java", "org.springframework.boot.loader.launch.JarLauncher"]
EXPOSE 8080

```

Elena-Bruyako commented on .gitignore:
```diff
@@ -31,3 +31,6 @@ build/
 .vscode/
+
+### database credentials ###
+.env*
```
provide `.env.template` file with key values

Elena-Bruyako commented on docker-compose.yml:
```diff
@@ -0,0 +1,54 @@
+    platform: linux/amd64
+    restart: unless-stopped
+    ports:
+      - "3307:3306"
```
use env variables

Elena-Bruyako commented on docker-compose.yml:
```diff
@@ -0,0 +1,54 @@
+    env_file:
+      - ./.env
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:mysql://db:3306/${MYSQLDB_DATABASE}?useSSL=false&allowPublicKeyRetrieval=true&serverTimezone=UTC&characterEncoding=UTF-8
```
use env variable instead of 3006 port

Elena-Bruyako commented on docker-compose.yml:
```diff
@@ -0,0 +1,54 @@
+      SPRING_JPA_HIBERNATE_DDL_AUTO: none
+      SPRING_JPA_PROPERTIES_HIBERNATE_DIALECT: org.hibernate.dialect.MySQLDialect
+      SPRING_LIQUIBASE_ENABLED: "true"
+      SPRING_LIQUIBASE_CHANGE_LOG: classpath:db/changelog/db.changelog-master.yaml
```
```suggestion
```

Elena-Bruyako commented on docker-compose.yml:
```diff
@@ -0,0 +1,54 @@
+      SPRING_JPA_PROPERTIES_HIBERNATE_DIALECT: org.hibernate.dialect.MySQLDialect
+      SPRING_LIQUIBASE_ENABLED: "true"
+      SPRING_LIQUIBASE_CHANGE_LOG: classpath:db/changelog/db.changelog-master.yaml
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
use DEBUG_PORT instead of 5005

Elena-Bruyako commented on docker-compose.yml:
```diff
@@ -0,0 +1,54 @@
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
+    ports:
+      - "8088:8080"
+      - "5005:5005"
```
use env variables

Elena-Bruyako commented on src/main/resources/application.properties:
```diff
@@ -1,15 +1,17 @@
-server.port=8080
-spring.security.user.name=Peter
-spring.security.user.password=777
+spring.datasource.url=${SPRING_DATASOURCE_URL}
```
add `spring.config.import=optional:file:.env[.properties]`

Elena-Bruyako reviewed (changes requested):
Have you tested your app in Docker? 
see comments

misterSokor committed:
- Revised Dockerfile to improve image build process
- Updated docker-compose.yml with corrected service configs and env references
- Adjusted application.properties to use env-based datasource settings

Elena-Bruyako reviewed (changes requested):
fix previous comments

misterSokor committed:
- Updated Dockerfile
- Added env.template file with key values
- Updated docker-compose.yml to use env variables (ports, DB configs, DEBUG_PORT) and fixed previous comments

sarakhmen commented on src/main/resources/application.properties:
```diff
@@ -1,15 +1,20 @@
-jwt.expiration-time=604800000
-jwt.secret=UnbreakableSecureKeyForAccessControl2025
+
+spring.jpa.properties.hibernate.dialect=${SPRING.JPA.PROPERTIES.HIBERNATE.DIALECT}
+spring.jpa.hibernate.ddl-auto=${SPRING_JPA_HIBERNATE_DDL_AUTO}
+
+spring.liquibase.enabled=${SPRING_LIQUIBASE_ENABLED}
+spring.liquibase.change-log=${SPRING_LIQUIBASE_CHANGELOG}
```
as far as these placeholders are not present in docker-compose , please provide corresponding values directly here

sarakhmen reviewed (changes requested):
.env.template is not present, please be more attentive to the previous comments

misterSokor committed:
- added missing .env.template file
- removed unused placeholders in application.properties and set explicit values
- aligned docker-compose.yml with env variables
